src/utils_general/raster_manipulation.py

```diff
@@ -99,111 +100,88 @@ def fix_calendar(ds, timevar="F"):
-    # the center inside the polygon are counted.
-    df["max_cell"] = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=raster_array,
-            affine=raster_transform,
-            band=band,
-            nodata=nodata,
-        )
-    )["max"]
-    df["max_cell_touched"] = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=raster_array,
-            affine=raster_transform,
-            all_touched=True,
-            band=band,
-            nodata=nodata,
-        )
-    )["max"]
-
-    df["avg_cell"] = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=raster_array,
-            affine=raster_transform,
-            band=band,
-            nodata=nodata,
-        )
-    )["mean"]
-    df["avg_cell_touched"] = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=raster_array,
-            affine=raster_transform,
-            all_touched=True,
-            band=band,
-            nodata=nodata,
-        )
-    )["mean"]
-
-    # calculate the percentage of the area within an geographical area
-    # that has a value larger than threshold
-    forecast_binary = np.where(raster_array >= threshold, 1, 0)
-    bin_zonal = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=forecast_binary,
-            affine=raster_transform,
-            stats=["count", "sum"],
-            band=band,
-            nodata=nodata,
-        )
-    )
-    df["perc_threshold"] = bin_zonal["sum"] / bin_zonal["count"] * 100
-    bin_zonal_touched = pd.DataFrame(
-        zonal_stats(
-            vectors=df,
-            raster=forecast_binary,
-            affine=raster_transform,
-            all_touched=True,
-            stats=["count", "sum"],
-            band=band,
-            nodata=nodata,
-        )
-    )
-    df["perc_threshold_touched"] = (
-        bin_zonal_touched["sum"] / bin_zonal_touched["count"] * 100
-    )
-
-    return df
+    df_list = []
+
+    for a in gdf[bound_col].unique():
+        gdf_adm = gdf[gdf[bound_col] == a]
+
+        da_clip = raster_array.rio.set_spatial_dims(
+            x_dim=lon_coord, y_dim=lat_coord
+        ).rio.clip(gdf_adm[geom_col], all_touched=all_touched)
+        if stats_list is None:
+            stats_list = ["mean", "std", "min", "max", "sum", "count"]
+
+        grid_stat_all = []
+        for s in stats_list:
+            if s == "count":
+                # count automatically ignores NaNs
+                # therefore skipna can also not be given as an argument
+                # implemented count cause needed for computing percentages
+                grid_stat = getattr(da_clip, s)(dim=[lon_coord, lat_coord])
+            else:
+                # if array only contains NaNs, "sum" will return 0
+                # while NaN might be preferred
+                # this can be fixed by setting the min_count arg
```

Comment:
If you have `kwargs` as a separate variable, you could add `min_count` as a key only if `s == "sum"`